python/tests/catalog/test_rest.py

```diff
@@ -0,0 +1,670 @@
+        message = "Invalid client ID: abc"
+        m.post(
+            f"{TEST_HOST}oauth/tokens",
+            json={
+                "error": {
+                    "message": message,
+                    "type": "BadCredentialsException",
+                    "code": 401,
+                }
+            },
+            status_code=401,
+        )
+
+        with pytest.raises(BadCredentialsError) as e:
+            RestCatalog("rest", {}, TEST_HOST, TEST_TOKEN)
+        assert message in str(e.value)
+
+
+def _configure_token_and_config_endpoint(m: requests_mock.Mocker):
+    m.post(
+        f"{TEST_HOST}oauth/tokens",
+        json={
+            "access_token": "eyJ0eXAiOiJK",
+            "token_type": "Bearer",
+            "expires_in": 86400,
+            "warehouse_id": "8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+            "region": "us-west-2",
+            "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+        },
+        status_code=200,
+    )
+    m.get(
+        f"{TEST_HOST}config",
+        json={"defaults": {}, "overrides": {"prefix": "oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e"}},
+        status_code=200,
+    )
+
+
+def test_list_tables_200():
+    with requests_mock.Mocker() as m:
+        _configure_token_and_config_endpoint(m)
+        namespace = "examples"
+        m.get(
+            f"{TEST_HOST}oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e/namespaces/{namespace}/tables",
+            json={"identifiers": [{"namespace": ["examples"], "name": "fooshare"}]},
+            status_code=200,
+        )
+
+        assert RestCatalog("rest", {}, TEST_HOST, TEST_TOKEN).list_tables(namespace) == [("examples", "fooshare")]
+
+
+def test_list_tables_404():
+    with requests_mock.Mocker() as m:
+        _configure_token_and_config_endpoint(m)
+        namespace = "examples"
+        m.get(
+            f"{TEST_HOST}oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e/namespaces/{namespace}/tables",
+            json={
+                "error": {
+                    "message": "Namespace does not exist: personal in warehouse 8bcb0838-50fc-472d-9ddb-8feb89ef5f1e",
+                    "type": "NoSuchNamespaceException",
+                    "code": 404,
+                }
+            },
+            status_code=404,
+        )
+        with pytest.raises(NoSuchTableError) as e:
+            RestCatalog("rest", {}, TEST_HOST, TEST_TOKEN).list_tables(namespace)
+        assert "Namespace does not exist" in str(e.value)
+
+
+def test_list_namespaces_200():
+    with requests_mock.Mocker() as m:
+        _configure_token_and_config_endpoint(m)
+        m.get(
+            f"{TEST_HOST}oss/warehouses/8bcb0838-22vo-472d-1925-8feb89lidf1e/namespaces",
+            json={"namespaces": [["default"], ["examples"], ["fokko"], ["system"]]},
+            status_code=200,
+        )
+        assert RestCatalog("rest", {}, TEST_HOST, TEST_TOKEN).list_namespaces() == [
```

Comment:
Most tests don't need to use `TEST_TOKEN`, right? If it isn't specific to what you're testing, I'd prefer to leave it out to keep code simpler.

Reply:
Removing this and the unnecessary prefix would mean you no longer need `_configure_token_and_config_endpoint`.

Reply:
When initializing the rest-catalog, we fetch the token and credential right away, therefore we need to set those.

Reply:
But if there are no credentials, we should skip the fetch token call, right?

Reply:
I'm sorry, I meant config instead of credentials. I've created a small fixture for this:
```python
@pytest.fixture
def rest_mock(requests_mock: Mocker):
    """Takes the default requests_mock and adds the config endpoint to it

    This endpoint is called when initializing the rest catalog
    """
    requests_mock.get(
        f"{TEST_HOST}v1/config",
        json={"defaults": {}, "overrides": {}},
        status_code=200,
    )
    return requests_mock
```
Regardless if we use credentials or an existing token, we should fetch the config (this contains information such as the prefix, that we need for subsequent calls).